homing_missiles.py

```diff
@@ -123,21 +164,95 @@ def draw_target_tags(self, screen):
                                      (projectile.position_[1]-projectile.size_/2) * self.DRAWING_SCALE_,
                                      projectile.size_ * self.DRAWING_SCALE_,
                                      projectile.size_ * self.DRAWING_SCALE_)
-            pygame.draw.rect(screen, self.projectile_color_, projectile_rect)
+            pygame.draw.rect(screen, self.enemy_color_, projectile_rect)
 
-        main_character = self.universe_.main_character_
-        main_character_rect = pygame.Rect((main_character.position_[0]-main_character.size_/2) * self.DRAWING_SCALE_,
-                                          (main_character.position_[1]-main_character.size_/2) * self.DRAWING_SCALE_,
-                                          main_character.size_ * self.DRAWING_SCALE_,
-                                          main_character.size_ * self.DRAWING_SCALE_)
-        pygame.draw.rect(screen, self.main_character_color_, main_character_rect)
+    def draw_targets(self, screen):
+        for enemy in self.targeted_enemies_: 
+            enemy_rect = pygame.Rect((enemy.position_[0]-enemy.size_/2) * self.DRAWING_SCALE_,
+                                     (enemy.position_[1]-enemy.size_/2) * self.DRAWING_SCALE_,
+                                     enemy.size_ * self.DRAWING_SCALE_,
+                                     enemy.size_ * self.DRAWING_SCALE_)
+
+            pygame.draw.rect(screen, self.target_color_, enemy_rect, 4)
+
+    """
+    Listener Functions
+    """
+
+    def reported_destroyed(self, game_object):
+        for enemy in self.targeted_enemies_:
+            if enemy == game_object:
+                self.targeted_enemies_.pop(self.targeted_enemies_.index(enemy))
     
 
-class HomingMissiles():
+class HomingMissilesProjectile(GameObject): 
+
+    def __init__(self, initial_position, enemy, universe):
+        self.position_ = initial_position
+        self.targeted_enemy_ = enemy
+        self.universe_ = universe
+        
+        self.ID_ = self.create_ID()
+        self.listeners_ = []
+        self.size_ = 5
+        self.speed_ = 5
+        self.velocity_ = self.calculate_trajectory(self.position_, self.targeted_enemy_.position_)
+        
+    def calculate_trajectory(self, initial_position, target_position):
+        x_distance = target_position[0] - initial_position[0]
+        y_distance = target_position[1] - initial_position[1]
+        distance = math.sqrt(x_distance**2 + y_distance**2)
+        
+        x_velocity = (x_distance * self.speed_) / distance
+        y_velocity = (y_distance * self.speed_) / distance
 
+        return x_velocity, y_velocity
+   
+    def check_collisions(self):
+        collisions = self.universe_.get_collisions(self)
+        for enemies in collisions:
+            enemies.take_damage(1000)
+
+        if collisions:
+            self.report_destroyed()
+
+    def update(self,events):
+        self.check_collisions()
```

Comment:
I'd recommend here, for the projectile to determine whether it's target has been destroyed. If it's target has been destroyed, it should do a report_destroyed()